src/async_fuse/memfs/s3_metadata.rs

```diff
@@ -258,32 +262,14 @@ impl<S: S3BackEnd + Sync + Send + 'static> MetaData for S3MetaData<S> {
                 size.cast()
             };
 
-        // let node_type = node.get_type();
-        // let file_data = if SFlag::S_IFREG == node_type {
         if inode.need_load_file_data(offset.cast(), size.cast()).await {
-            debug!("read() need to load file data of ino={}", ino,);
-            let load_res = inode.load_data(offset.cast(), size.cast()).await;
-            if let Err(e) = load_res {
-                debug!(
-                    "read() failed to load file data of ino={} and name={:?}, the error is: {:?}",
-                    ino,
-                    inode.get_name(),
-                    e,
-                );
-                return Err(e);
-            }
+            inode.load_data(offset.cast(), size.cast()).await?;
         }
         return Ok(inode.get_file_data(offset.cast(), size.cast()).await);
     }
 
     #[instrument(skip(self), err, ret)]
     async fn open(&self, context: ReqContext, ino: u64, flags: u32) -> DatenLordResult<RawFd> {
-        let node = self
-            .get_node_from_kv_engine(ino)
-            .await?
-            .ok_or_else(|| build_inconsistent_fs!(ino))?;
-        let o_flags = fs_util::parse_oflag(flags);
-        node.open_pre_check(o_flags, context.user_id, context.group_id)?;
         // TODO: handle open flags
         // <https://pubs.opengroup.org/onlinepubs/9699919799/functions/open.html>
```

Comment:
Do we still need these comments?

Reply:
We might use this later when we build the open-to-close cache.